src/request_handler.rs

```diff
@@ -92,63 +147,18 @@ pub(crate) mod tests {
 
   #[test]
   fn index_route_status_code_is_200() {
-    test(|port, _dir| async move {
-      assert_eq!(
-        reqwest::get(format!("http://localhost:{}", port))
-          .await
-          .unwrap()
-          .status(),
-        200
-      )
-    });
+    test(|url, _dir| async move { assert_eq!(reqwest::get(url).await.unwrap().status(), 200) });
```

Comment:
I noticed some calls to `as_str` that were mostly not necessary, so I removed them.